### ISSUE TITLE
Can't compile on latest nightly?

### ISSUE THREAD

**Issue by mre** [jkarns275/stdgba#3]
Rust version: `rustc 1.27.0-nightly (2f2a11dfc 2018-05-16)`
last stdgba commit: https://github.com/jkarns275/stdgba/commit/d46b87bb218b8e847bc5d0130e9e563ba6e1f318

When I run `cargo build`, I get the following error:

```rust
cargo build                                                                                       ✘ 130
   Compiling g v0.1.0
   Compiling gbaimg v0.1.0
...

error[E0512]: transmute called with types of different sizes
   --> gbaimg/src/lib.rs:192:34
    |
192 |             let trans = unsafe { transmute::<proc_macro::Literal, token::Token>(x.clone()) };
    |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: source type: proc_macro::Literal (160 bits)
    = note: target type: syntax::parse::token::Token (192 bits)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0512`.
error: Could not compile `gbaimg`.
warning: build failed, waiting for other jobs to finish...
```

Might well be that I'm misunderstanding something but it seems like the memory layout of `proc_macro::Literal` or `syntax::parse::token::Token` has changed in a recent version.
Which version of those crates to you have in your `Cargo.lock`?

**Comment by jkarns275** [comment]
Strange. Are you using xargo? Not sure if that is the cause of the issue but you do need to use xargo rather than normal cargo. I'll take a look at this later today or tomorrow though. Thanks for the bug report!

**Comment by mre** [comment]
Oh wow, here's another attempt of making a complete fool of myself: I didn't use xargo. 😆 I'm sorry, will try that now.

**Comment by jkarns275** [comment]
@mre any update on this?

**Comment by jkarns275** [comment]
I've fixed it in 60147eb9fdebe2bc037a4b2410d88d299892d94b. The gbaimg crate was littered with bugs! I wasn't taking advantage of the `syn` crate to make the use of proc_macro input easier, and I had only half-implemented a feature and totally forgot what the feature was supposed to do. It should however work now. I'll have to test to make sure the gbaimg library works properly now, but for now I'm going to close this issue since I'm pretty certain it is fixed - at the very least it compiles :)!

**Comment by mre** [comment]
Yeah, wanted to write a follow-up on that, but got busy. Anyway, compiles now just fine. Thanks. 👍
